src/driver.ts

```diff
@@ -153,24 +153,76 @@ export abstract class CMakeDriver implements vscode.Disposable {
+
+    // Update default replacements
+    replacements['workspaceRoot'] = vscode.workspace.rootPath;
+    replacements['buildType'] = this.currentBuildType;
+    replacements['workspaceRootFolderName'] = path.basename(ws_root);
+    replacements['generator'] = this.generatorName || 'null';
+    replacements['projectName'] = this.projectName;
+    replacements['userHome'] = user_dir;
+
+    // Update Variant replacements
+    const variantSettings = this.stateManager.activeVariantSettings;
+    if (variantSettings) {
+      variantSettings.forEach((value: string, key: string) => {
+        if (key != 'buildType') {
+          replacements[key] = value;
+        } else {
+          replacements['buildLabel'] = value;
+        }
+      });
+    }
+
+    this.stateManager.replacements = replacements;
+
+    return replacements;
+  }
+
+  /**
+   * Get the environment and apply any needed
+   * substitutions before returning it.
+   */
+  async getExpandedEnvironment(): Promise<{[key: string]: string}> {
+    const env = {} as {[key: string]: string};
+    await Promise.resolve(util.objectPairs(config.environment)
+                              .forEach(async ([key, value]) => env[key] = await this.expandString(value)));
+    return env;
+  }
+
+  /**
+   * Get the configure environment and apply any needed
+   * substitutions before returning it.
+   */
+  async getExpandedConfigureEnvironment(): Promise<{[key: string]: string}> {
+    const config_env = {} as {[key: string]: string};
+    await Promise.resolve(util.objectPairs(config.configureEnvironment)
+                              .forEach(async ([key, value]) => config_env[key] = await this.expandString(value)));
+    return config_env;
+  }
+
   /**
    * Replace ${variable} references in the given string with their corresponding
    * values.
    * @param instr The input string
    * @returns A string with the variable references replaced
    */
-  async expandString(instr: string): Promise<string> {
-    const ws_root = util.normalizePath(vscode.workspace.rootPath || '.');
-    type StringObject = {[key: string]: string | undefined};
-    const user_dir = process.platform === 'win32' ? process.env['PROFILE']! : process.env['HOME']!;
-    const replacements: StringObject = {
-      workspaceRoot: vscode.workspace.rootPath,
-      buildType: this.currentBuildType,
-      workspaceRootFolderName: path.basename(ws_root),
-      generator: this.generatorName || 'null',
-      projectName: this.projectName,
-      userHome: user_dir,
-    };
+  async expandString(instr: string, env?: proc.EnvironmentVariables): Promise<string> {
+    // Update the replacements and get the updated values
+    const replacements = this._replacements;
+
+    // Setup temporary environment
+    const old_env = process.env;
+    const custom_env = util.mergeEnvironment(old_env as proc.EnvironmentVariables, config.environment, env ? env : {});
+    process.env = custom_env;
```

Comment:
I may be missing some context, but why do we modify the extension's own environment variables?

Reply:
I added an optional parameter to the `expandString()` method, which gets applied there, and reset later. But you are right, I could probably just do the regex stuff directly with the passed environment, without modifying the actual process one.

Reply:
Done.